ddprof-ffi/src/exporter.rs

```diff
@@ -122,27 +46,21 @@ pub unsafe extern "C" fn buffer_reset(buffer: *mut Buffer) {
     }
 }
 
-/// Destroys the Exporter.
-#[export_name = "ddprof_ffi_Exporter_delete"]
-pub extern "C" fn exporter_delete(exporter: Option<Box<Exporter>>) {
-    std::mem::drop(exporter)
-}
-
 #[repr(C)]
 pub struct Tag<'a> {
-    name: ByteSlice<'a>,
+    name: CharSlice<'a>,
     value: ByteSlice<'a>,
 }
 
 #[repr(C)]
 pub enum EndpointV3<'a> {
-    Agent(ByteSlice<'a>),
-    Agentless(ByteSlice<'a>, ByteSlice<'a>),
+    Agent(CharSlice<'a>),
+    Agentless(CharSlice<'a>, CharSlice<'a>),
 }
```

Comment:
I noticed this broke one of the FFI examples:

```
/Users/ivo.anjo/datadog/x-second-libddprof/examples/ffi/exporter.cpp:109:38: error: no matching function for call to
      'ddprof_ffi_EndpointV3_agentless'
    ddprof_ffi_EndpointV3 endpoint = ddprof_ffi_EndpointV3_agentless(to_byteslice("datad0g.com"), to_byteslice(api_key));
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ivo.anjo/datadog/x-second-libddprof/build-macos/include/ddprof/ffi.h:329:30: note: candidate function not viable: no known
      conversion from 'ddprof_ffi_ByteSlice' (aka 'ddprof_ffi_Slice_u8') to 'ddprof_ffi_CharSlice' (aka 'ddprof_ffi_Slice_c_char')
      for 1st argument
struct ddprof_ffi_EndpointV3 ddprof_ffi_EndpointV3_agentless(ddprof_ffi_CharSlice site,
                             ^
/Users/ivo.anjo/datadog/x-second-libddprof/examples/ffi/exporter.cpp:110:31: error: no viable conversion from
      'ddprof_ffi_ByteSlice' (aka 'ddprof_ffi_Slice_u8') to 'const char *'
    ddprof_ffi_Tag tags[] = {{to_byteslice("service"), to_byteslice(service)}};
                              ^~~~~~~~~~~~~~~~~~~~~~~
/Users/ivo.anjo/datadog/x-second-libddprof/examples/ffi/exporter.cpp:110:56: error: no viable conversion from
      'ddprof_ffi_ByteSlice' (aka 'ddprof_ffi_Slice_u8') to 'size_t' (aka 'unsigned long')
    ddprof_ffi_Tag tags[] = {{to_byteslice("service"), to_byteslice(service)}};
                                                       ^~~~~~~~~~~~~~~~~~~~~
/Users/ivo.anjo/datadog/x-second-libddprof/examples/ffi/exporter.cpp:124:17: error: no viable conversion from
      'ddprof_ffi_ByteSlice' (aka 'ddprof_ffi_Slice_u8') to 'const char *'
        .name = to_byteslice("auto.pprof"),
```

One thing I've been thinking is that it would perhaps make sense to build the examples when we run `ffi-build.sh`, to avoid bitrotting them (and as an extra "test" that API changes are deliberate).

Reply:
Interesting; I built this example and it was fine. Now I need to figure out where I screwed up ^_^